### PR TITLE
Foreign Keys on zeder_journals.id

### PR DESCRIPTION
**Pull request by mtrojan-ub**
Merge needs to wait until zeder_journals has been populated on nu+ub28

### CODE REVIEW

File: cpp/lib/src/ZoteroHarvesterUtil.cc [hunk 1]
```diff
@@ -493,11 +499,12 @@ std::vector<UploadTracker::Entry> UploadTracker::getEntriesByZederIdAndFlavour(c
     DbConnection db_connection;
 
     const std::string zeder_instance(GetZederInstanceString(zeder_flavour));
-    db_connection.queryOrDie("SELECT t2.url, t1.delivered_at, t1.zeder_id, t1.zeder_instance, t1.main_title, t1.hash "
-                             "FROM delivered_marc_records_urls as t2 "
-                             "LEFT JOIN delivered_marc_records as t1 ON t2.record_id = t1.id WHERE t1.zeder_id="
-                             + db_connection.escapeString(std::to_string(zeder_id)) + " AND t1.zeder_instance="
-                             + db_connection.escapeAndQuoteString(zeder_instance));
+    db_connection.queryOrDie("SELECT t2.url, t1.delivered_at, t0.zeder_id, t0.zeder_instance, t1.main_title, t1.hash "
```
Comment (ruschein) [inline cpp/lib/src/ZoteroHarvesterUtil.cc:502]: And again.

File: cpp/data/installer/ub_tools.sql [hunk 2]
```diff
@@ -19,42 +21,40 @@ CREATE TABLE rss_aggregator (
...
 -- Table to be used w/ our validate_harvested_records tool:
 CREATE TABLE metadata_presence_tracer (
-    journal_name VARCHAR(764) NOT NULL,
+    zeder_journal_id INT(11) UNSIGNED NOT NULL,
     metadata_field_name CHAR(4) NOT NULL,
     field_presence ENUM('always', 'sometimes', 'ignore') NOT NULL,
-    UNIQUE(journal_name, metadata_field_name),
-    INDEX journal_name_and_metadata_field_name_index(journal_name, metadata_field_name)
+    CONSTRAINT metadata_presence_tracer_zeder_journal_id FOREIGN KEY (zeder_journal_id) REFERENCES zeder_journals (id) ON DELETE CASCADE ON UPDATE CASCADE,
+    CONSTRAINT metadata_presence_tracer_journal_field UNIQUE (zeder_journal_id, metadata_field_name)
```
[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply cpp/data/installer/ub_tools.sql:45]: metadata_presence_tracer_zeder_journal_id_and_metadata_field_name

That is too long for MySQL

[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply cpp/data/installer/ub_tools.sql:45]: So i will change it to metadata_presence_tracer_journal_id_and_field_name

File: cpp/data/installer/ub_tools.sql [hunk 3]
```diff
@@ -4,10 +4,12 @@
 -- diacritical marks => we need to override it and use utf8mb4_bin.
 
 -- The sizes here must be in sync with the constants defined in rss_aggregator.cc!
+
+-- Please always add names to your CONSTRAINTs! (for succeeding sql_updates)
```
[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply cpp/data/installer/ub_tools.sql:8]: done

File: cpp/lib/src/ZoteroHarvesterUtil.cc [hunk 4]
```diff
@@ -370,9 +370,11 @@ static void UpdateUploadTrackerEntryFromDbRow(const DbRow &row, UploadTracker::E
 bool UploadTracker::urlAlreadyDelivered(const std::string &url, Entry * const entry,
                                         DbConnection * const db_connection) const
 {
-    db_connection->queryOrDie("SELECT t2.url, t1.delivered_at, t1.zeder_id, t1.zeder_instance, t1.main_title, t1.hash "
-                              "FROM delivered_marc_records_urls as t2 "
-                              "LEFT JOIN delivered_marc_records as t1 ON t2.record_id = t1.id WHERE t2.url='"
+    db_connection->queryOrDie("SELECT t2.url, t1.delivered_at, t0.zeder_id, t0.zeder_instance, t1.main_title, t1.hash "
```
[the remark this answers is not in the document]
Comment (mtrojan-ub) [reply cpp/lib/src/ZoteroHarvesterUtil.cc]: Are the changed shorthands better? (t0=>zj for zeder_journals, t1 => dmr for delivered_marc_records, t2 => dmru for delivered_marc_records_urls)

If i completely remove them, the statements will be a lot longer...

**Pull request merged**